transforms/csdl/preprocess_csdl.xsl

```diff
@@ -487,6 +487,31 @@
         <xsl:attribute name="Type">Collection(graph.conditionalAccessConditions)</xsl:attribute>
     </xsl:template>
     
+    <!--Delta function for events need the start and end date parameters-->
+    <xsl:template match="edm:Schema[@Namespace='microsoft.graph']/edm:Function[@Name='delta'][edm:Parameter[@Name='bindingparameter']][edm:Parameter[@Type='Collection(graph.event)']]">
+        <xsl:copy>
+            <xsl:apply-templates select="@* | node()"/>
+            <Annotation Term="Org.OData.Capabilities.V1.OperationRestrictions">
+            <Record>
+                <PropertyValue Property="CustomQueryOptions">
+                    <Collection>
+                        <Record>
+                            <PropertyValue Property="Name" String="startDateTime" />
```

Comment:
Should we set the type as well?

Reply:
I also thought about this as well, but when I was digging through this, what I found was that a CustomParameter is always a string. Validation can be enforced but there is no type of specification. 

- https://github.com/microsoft/OpenAPI.NET.OData/blob/f13fd9a453849128df87bba0e5c2ce8cb46a756e/src/Microsoft.OpenApi.OData.Reader/Vocabulary/Capabilities/CustomParameter.cs#L18
- https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Capabilities.V1.md#customparameter